paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/reporting/ClassPageRenderer.kt

```diff
@@ -0,0 +1,208 @@
+package app.cash.paparazzi.gradle.reporting
+
+import org.gradle.api.internal.tasks.testing.junit.result.TestResultsProvider
+import org.gradle.api.tasks.testing.TestOutputEvent
+import org.gradle.internal.html.SimpleHtmlWriter
+import org.gradle.internal.xml.SimpleMarkupWriter
+import org.gradle.reporting.CodePanelRenderer
+import java.io.File
+import java.io.IOException
+import kotlin.io.encoding.Base64
+import kotlin.io.encoding.ExperimentalEncodingApi
+
+internal class ClassPageRenderer(
+  private val resultsProvider: TestResultsProvider,
+  isVerifyRun: Boolean,
+  failureDir: File
+) : PageRenderer<ClassTestResults>() {
+  private val codePanelRenderer = CodePanelRenderer()
+  private val imagePanelRenderer = ImagePanelRenderer()
+
+  @OptIn(ExperimentalEncodingApi::class)
+  private val diffImages: List<DiffImage> =
+    if (isVerifyRun && failureDir.exists()) {
+      failureDir.listFiles()
+        ?.filter { it.name.startsWith("delta-") }
+        ?.map { diff ->
+          // TODO: read from failure diff metadata file instead of brittle parsing
+          val nameSegments = diff.name.split("_", limit = 3)
+          val testClassPackage = nameSegments[0].replace("delta-", "")
+          val testClass = "$testClassPackage.${nameSegments[1]}"
+          val testMethodWithLabel = nameSegments[2].split(".")[0]
```

Comment:
nit: this misses some snapshots. I have a fix for this here: https://github.com/cashapp/paparazzi/pull/1677

Reply:
Lets get this in first, then I can update my branch.

Reply:
Cherry-picked your [commit](https://github.com/cashapp/paparazzi/pull/1675/commits/aaba36d1a53fb6eae521acfec905edd62ac03eaf) for attribution with a little [cleanup](https://github.com/cashapp/paparazzi/pull/1675/commits/badc30eb17bfdfe2f9494c69cf6b5a64bc86f6ce).